### PR TITLE
map.c: optimize dbscan code

### PR DESCRIPTION
**Pull request by ralfbrown**
Yields ~25% speedup, which becomes noticeable when there are >50k images being clustered for the map view.  (The clustering runtime is not quite quadratic -- 55k points goes from 0.8 seconds to 0.55, while 359k goes from 25.9 sec to 18.8.)

### CODE REVIEW

File: src/views/map.c [hunk 1]
```diff
@@ -2879,7 +2883,7 @@ static void _dbscan_spread(unsigned int index)
   for(unsigned int i = 0; i < db.spreads->num_members; i++)
   {
     dt_geo_position_t *d = &db.points[db.spreads->index[i]];
-    if(d->cluster_id == NOISE || d->cluster_id == UNCLASSIFIED)
+//    if(d->cluster_id < 0) // NOISE or UNCLASSIFIED - but only those are included in the list anyway
```
Comment (TurboGit) [inline src/views/map.c:2886]: Probably to be removed.

Comment (TurboGit) [reply src/views/map.c:2886]: @ralfbrown : Can you comment on this? Should it be removed or is it a left over of some debug session?

I can remove it myself if needed, but I prefer to ask. TIA.

Comment (ralfbrown) [reply src/views/map.c:2886]: Sorry, missed the comment previously.  I can do it over the weekend unless you're in a hurry....

Comment (TurboGit) [reply src/views/map.c:2886]: Nothing urgent, I'll wait for the weekend. Thanks!

Comment (ralfbrown) [reply src/views/map.c]: Done.  Also rebased.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**